src/index.ts

```diff
@@ -19,52 +18,24 @@ const run: Edkt['run'] = async (config) => {
     if (!hasConsent) return;
   }
 
-  // This is a no-op if undefined, equals current value or lesser than 0
   viewStore.setStorageSize(featureStorageSize);
-  viewStore.insert(pageFeatures, pageMetadata);
+  viewStore.savePageViews(pageFeatures, pageMetadata);
 
-  const matchedAudiences = audienceDefinitions
-    .filter((audience) => {
-      return !matchedAudienceStore.matchedAudienceIds.includes(audience.id);
-    })
-    .map((audience) => {
-      return {
-        ...audience,
-        conditions: engine.translate(audience),
-      };
-    })
-    .map((audience) => {
-      const currentTS = timeStampInSecs();
-      const pageViewsWithinLookBack = viewStore.pageViews.filter((pageView) => {
-        return audience.lookBack === 0
-          ? true
-          : pageView.ts > currentTS - audience.lookBack;
-      });
-      return {
-        id: audience.id,
-        matchedAt: currentTS,
-        expiresAt: currentTS + audience.ttl,
-        matchedOnCurrentPageView: true,
-        matched: engine.check(audience.conditions, pageViewsWithinLookBack),
-      };
-    })
-    .filter((audience) => audience.matched);
+  const pageViews = audienceDefinitions.map(({ lookBack }) =>
+    viewStore.getPageViewsWithinLookBack(lookBack)
+  );
```

Comment:
Since you are calling this once for each audience, you are going to duplicate the page views many times, if you have 20 audiences, 200 views - than this will return 4k*128dims array which is pretty large.

Reply:
Because of coupling.
Either you have to pass the store to the engine to query it, or you have to query the `pageViews` outside the engine and pass it onto.
Maybe there is a third way I'm missing?

Reply:
Why can we not pass a single copy of all the pageviews to `engine.getMatchingAudiences`, then inside this module for each audience, take the correct slice of pageviews before calling engine.check?